### ISSUE TITLE
[16.0][BUG] Missing multi-company rules on rating

### ISSUE THREAD

**Issue by carolinafernandez-tecnativa** [odoo/odoo#163001]
**Impacted versions:**
15.0, 16.0, 17.0

**Steps to reproduce:**

1- Multi-environment configuration
2- Allow customer ratings on project.
3- Go to Project > Reporting > Customer ratings
4- All customer ratings belong to Company San Francisco.

<img width="1680" alt="image" src="https://github.com/odoo/odoo/assets/143796894/99b363a9-28b8-46f2-911f-b6cd1177ca72">

5- User is current on Company Chicago

<img width="1680" alt="image" src="https://github.com/odoo/odoo/assets/143796894/8c15b753-6a2b-411b-8be3-2dae5a06bade">

6- Tried to access one customer rating

<img width="1676" alt="image" src="https://github.com/odoo/odoo/assets/143796894/660e7bf8-30c3-4c33-b653-5192cc6cc11b">

**Current behavior:**

If I have multi-company environment, all customer ratings tasks are not being filtered by company, it shows all ratings for all companies, independently which company is the user at.

**Expected behavior:**

If I have multi-company environment, when i go to check customer ratings task as an example, it has to show only task rating for the company that user is logged in.
In this example, when the user is with Company Chicago, when access to Customer ratings, it should be an empty list.

**Please note, there is no field company_id on model rating.rating and security rule is missing.**

**Comment by pedrobaeza** [comment]
@mart-e do you know who is in charge of this module?

**Comment by vava-odoo** [comment]
That would be something for @xavierbol. Could you have a look?

**Comment by pedrobaeza** [comment]
I think this can be implemented overriding `_check_access` and checking if the user have access to the rating related record. This way, it's not only multi-company, but any other restriction through record rules can be checked.

**Comment by pedrobaeza** [comment]
@xavierbol @vava-odoo any news?

**Comment by xavierbol** [comment]
Hello @pedrobaeza 

I will check. 🙂 

Sorry for the delay... 😢

**Comment by xavierbol** [comment]
Hello @carolinafernandez-tecnativa @pedrobaeza 

Thanks for your feedback and report. 🙂 

As you said the `company_id` field does not exist on `rating.rating` model and so we cannot really have a `multi-company` security rule. Because of that, we cannot really do the expected behavior that you mentioned since we cannot filter the ratings to be sure the company set on the related document is in the ones selected by the current user. (We could probably do it in master, I will create a task to discuss about that with my colleagues 🙂 ).

(to investigate) In stable, the solution, we could find, is to let the current user accesses to the rating form view even if the related document is from another company (or is not accessible by the user). I guess it could be acceptable to avoid having a potential traceback, saying the user cannot access to the related document, each time the user would like to open the form view of a rating.

What do you think? 🙂

**Comment by pedrobaeza** [comment]
So don't you think a `check_access_rule` or `_filter_access_rules_python` override checking the company of the related document and filtering according allowed companies can be an option?

**Comment by xavierbol** [comment]
@pedrobaeza 
I don't really like overriding those methods, TBH. 😅 

Also, I don't really see how your suggestion could work to filter the ratings displayed on the kanban and list views since those methods are not called by the search/search_read method (`check_access_rule` could indeed be called by `_read` but it will return a traceback in the case we should not see that record, it is not really used to filter the records to read). 🤔 
I maybe miss something. 🧐 

Moreover, I think we could also have that traceback if the user does not have access to a private project for instance. 🤷🏼‍♂️ 

Finally, those overrides could be very costly, IMO. 😕

**Comment by pedrobaeza** [comment]
Yes, it can be costly, as if we want to do it right for cases like private projects, we have to call `_check_access` record by record. Then do your best to mitigate the problem in stable versions.

For master, an initial approach can be to have a computed field `company_id` and put the record rule according that, but it will still be the problem with records not accessible (like private projects). Although you don't like it, maybe some overrides on access methods that call the equivalent on linked records to filter them out?

**Comment by xavierbol** [comment]
Actually, after investigation, no need to take care of the private projects since the `Customer Ratings` menu is only accessible by Project Manager, so, fortunately, in standard, we just got that traceback in multi-company. 🤔 😁

**Comment by pedrobaeza** [comment]
Great, you can ping me if you want me to test anything.
